src/components/builder.ts

```diff
@@ -597,18 +603,16 @@ export class Builder {
         }
     }
 
-    private findMagicPrograms(rootFile: string): [Tool | undefined, Tool | undefined] {
+    private findMagicComments(rootFile: string): {tex?: Tool, bib?: Tool, recipe?: string} {
```

Comment:
For the interface: I think we can keep it as is, since this type was not used elsewhere except for L448. We absolutely can make it a type if later this function is wider used.

Beginning of doc: That's a good idea. What about we define the comment block to be multiple consecutive lines starting with either a `%` or all spaces (empty)?

Reply:
> What about we define the comment block to be multiple consecutive lines starting with either a `/%\s*!/` or all spaces (empty)?

Yes. This looks a good definition of the first comment block.